### PR TITLE
Modified documents and deleted images

### PR DESCRIPTION
**Pull request by mgundala**
Deleted images and modified content as per suggestions

### CODE REVIEW

File: docs/bdd-introduction.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+## Behavior Driven Development(BDD):
+
+   Behavior driven development (or BDD) is an software development technique that encourages collaboration between developers, QA and non-technical or business participants in a software project.
```
Comment (vcavallo) [inline docs/bdd-introduction.md]: > is an software development

is a software development

> Behavior driven

Behavior-driven

File: docs/bdd-introduction.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+## Behavior Driven Development(BDD):
```
Comment (vcavallo) [inline docs/bdd-introduction.md]: > Behavior Driven

Behavior-Driven

File: docs/bdd-introduction.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+   Behavior driven development (or BDD) is an software development technique that encourages collaboration between developers, QA and non-technical or business participants in a software project.
+
+   BDD focuses on obtaining a clear understanding of desired software behavior through discussion with stakeholders. It extends [TDD](docs/tdd-introduction) by writing test cases in a natural language that non-programmers can read. Behavior-driven developers use their native language in combination with the ubiquitous language of domain driven design to describe the purpose and benefit of their code. This allow the developers to focus on why the code should be created, rather than the technical details, and minimizes translation between the technical language in which the code is written and the domain language spoken by the business, users, stakeholders, project management, etc.
```
Comment (vcavallo) [inline docs/bdd-introduction.md]: > domain driven design

domain-driven

File: docs/bdd-introduction.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+     
+   Coming to reality, Customer is a person who is an expert in dealing the things other than Technology. They need not to be aware of Edge case scenarios,beta testing and Alpha testing and so on. Even if he(customer) is familiar of the Technology stack,the only thing that is evolving in much faster pace is business, which change at constant pace.
```
Comment (vcavallo) [inline docs/bdd-introduction.md]: > Coming to reality, 

This part is a bit weird. not sure what to do with it.

## suggestion:

A _Customer_ is a stakeholder who is an expert at dealing directly with the domain, rather than the technology responsible for presenting the domain. The customer need not be aware of edge cases, the difference between alpha and beta testing, unit and feature tests, etc. Even if the customer is familiar with the evolving development process of the technology stack behind the user interface, he or she is typically more concerned with the domain and the business - which themselves are often changing at a rapid pace.

[ even with this suggested edit, the paragraph is still a bit strange ]

File: docs/bdd-introduction.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+     
+   In this modern world, teams understand the business change and structure their approach according to the business needs.Rather than attempt to conceive, build, and deliver the perfect application in one fell swoop, they steadily deliver work in small and usable chunks for every week continously.  The customer is encouraged to think in terms of what functionality will be delivered at the maximum benefit to the business and prioritize the pipeline of work accordingly.
```
Comment (vcavallo) [inline docs/bdd-introduction.md]: ## minor changes below:

In this modern world, teams understand the changes in their business and they structure their approach according to these needs. Rather than attempt to conceive, build, and deliver the perfect application in one fell swoop, they steadily deliver work in small and usable chunks every week, continuously.  The customer is encouraged to think in terms of what functionality will be delivered to the maximum benefit of the business and prioritize the pipeline of work accordingly.

File: docs/bdd-introduction.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+     
+   This is where BDD comes in.  It bakes quality into the process by smoothing communication between stakeholders.  
+  
+   **"Behavior-driven development is about implementing an application by describing its behavior from the perspective of its stakeholders”** -- Dan North
+   
+  The main intention behind BDD framework is to avoid communication gaps between team memebers, And have better understanding and communicating more often with real world examples.
```
Comment (vcavallo) [inline docs/bdd-introduction.md]: ## suggestion:

The main intention of a BDD framework is to avoid communication gaps between team members, foster better understanding of the customer and promote continuous communication with real world examples.

File: docs/tdd-introduction.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+
+
+## Test Driven Development
+
+Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle: requirements are turned into very specific test cases, then the software is improved to pass the new tests. This is opposed to software development that allows software to be added that isn't proven to meet requirements. Begin by writing a very small test for code that does not yet exist. Run the test and, naturally, it fails. Now write just enough code to make that test pass. Once the test passes, observe the resulting design and re-factor to remove any duplication you see. It is natural at this point to judge the design as too simple to handle all of the responsibilities this code will have.
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: > Once the test passes, observe the resulting design and re-factor to remove any duplication you see.

suggestion:

Once the test passes, observe the resulting design and re-factor according to whatever software design principles seem to best apply.

---

`DRY` isn't the only thing to pay attention to ;)

File: docs/tdd-introduction.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+Test Driven Development is a developer practice that involves writing tests before writing the code being tested. Begin by writing a very small test for code that does not yet exist. Run the test and, naturally, it fails. Now write just enough code to make that test pass. No more. Once the test passes, observe the
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: ~Test Driven Development is a developer practice that involves writing tests before writing the code being tested. Begin by writing a very small test for code that does not yet exist. Run the test and, naturally, it fails. Now write just enough code to make that test pass. No more. Once the test passes, observe the~

remove this entirely. it's a repetition of above.

File: docs/tdd-introduction.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+resulting design and re-factor to remove any duplication you see. It is natural at this point to judge the design as too simple to handle all of the responsibilities this code will have.
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: ~resulting design and re-factor to remove any duplication you see. It is natural at this point to judge the design as too simple to handle all of the responsibilities this code will have.~

again, remove this. already stated verbatim above.

File: docs/tdd-introduction.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+As the code base gradually increases in size, more and more attention is consumed by the re-factoring step. The design is constantly evolving and under constant review, though it is not predetermined. This process is known as emergent design, and is one of the most significant by products of Test Driven Development.
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: > This process is known as emergent design, and is one of the most significant by products of Test Driven Development. 

- `by-products` or `byproducts`
- `Test-Driven Development`

File: docs/tdd-introduction.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+Test Driven Development is a developer practice that involves writing tests before writing the code being tested. Begin by writing a very small test for code that does not yet exist. Run the test and, naturally, it fails. Now write just enough code to make that test pass. No more. Once the test passes, observe the
+resulting design and re-factor to remove any duplication you see. It is natural at this point to judge the design as too simple to handle all of the responsibilities this code will have.
+
+As the code base gradually increases in size, more and more attention is consumed by the re-factoring step. The design is constantly evolving and under constant review, though it is not predetermined. This process is known as emergent design, and is one of the most significant by products of Test Driven Development. 
+
+To get better understanding here we go life cycle of TDD
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: ## suggstion, simply:

The life-cycle of TDD:

File: docs/tdd-introduction.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+  * write a test
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: `Write` a test

File: docs/tdd-introduction.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+  * Run it ( No implementation code in place, Test need to fail)
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: > * Run it ( No implementation code in place, Test need to fail)

`* Run it (no implementation code in place, test needs to fail)`

File: docs/tdd-introduction.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+  * write implemnetation code to make the test pass
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: capitalize `Write`

File: docs/tdd-introduction.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+  * Run it
+  * Refactor the tset for better code quality
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: > tset

`test`

File: docs/tdd-introduction.md [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+  
+This procdure is called as **RED-GREEN-REFACTOR**
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: > This procdure is called as **RED-GREEN-REFACTOR**

This procedure is called **RED-GREEN-REFACTOR**

File: docs/tdd-introduction.md [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+
+Final Note: TDD is aot about testing, It is the process of approach towards your design and force us to think about the implemantation before we code.
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: > Final Note: TDD is aot about testing, It is the process of approach towards your design and force us to think about the implemantation before we code.

Final Note: TDD is not only about testing; it is the process of approaching your design and forcing you to think about the desired outcome and API before you code.

---

I would remove `implementation`... it's the opposite - you don't want to think about the _implementation_ at all before you code, rather, you want to think about the API and how you expect the final product to be interacted with.

File: docs/tdd-introduction.md [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+
+The main objective of this framework is code design with tests.
```
Comment (vcavallo) [inline docs/tdd-introduction.md]: > The main objective of this framework is code design with tests.

Remove this entirely, I think.

File: docs/bdd-introduction.md [hunk 19]
```diff
@@ -1,17 +1,16 @@
...
+    
+    Let me try to explain with an example, A client wanted to increase his business by developing an important feature to his software.The Client gathered all his stake holders to the meeting where they have discussions about the feature.Mostly the outcome of the discussion will be like [a blind man expalining an elephant](https://en.wikipedia.org/wiki/Blind_men_and_an_elephant). It's inevitable, however the Client reaches to one development team and explains what he really wanted. Sometimes these discussions take shape of user Stroies(in Agile world), these is the time where the developer will handover his hat to Client to describe about feature.After a succesfull description developer has his hat back on implementation.
+    
+    Only these communications are responsible for successfull implepmentation of client's feature.When a developer is using BDD to develop a feature, he speaks the language of client. And the cleint can easily understand what exactly is happening and how it has been implemented at any point of implementation. Ideally speaking they both( Developer and Client) collaboratively will design and develop the feature.
```
Comment (olleolleolle) [inline docs/bdd-introduction.md]: _Missing space before `When`_

File: docs/bdd-introduction.md [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+   **"Behavior-driven development is about implementing an application by describing its behavior from the perspective of its stakeholders”** -- Dan North
+   
+    The main intention of a BDD framework is to avoid communication gaps between team members, foster better understanding of the customer and promote continuous communication with real world examples.
+    
+    Let me try to explain with an example, A client wanted to increase his business by developing an important feature to his software.The Client gathered all his stake holders to the meeting where they have discussions about the feature.Mostly the outcome of the discussion will be like [a blind man expalining an elephant](https://en.wikipedia.org/wiki/Blind_men_and_an_elephant). It's inevitable, however the Client reaches to one development team and explains what he really wanted. Sometimes these discussions take shape of user Stroies(in Agile world), these is the time where the developer will handover his hat to Client to describe about feature.After a succesfull description developer has his hat back on implementation.
```
Comment (mlvandijk) [inline docs/bdd-introduction.md]: I'd like to suggest removing the last 2 paragraphs (from "Let me try to explain...") entirely. Imho the first paragraph is clear enough.

File: docs/tdd-introduction.md [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+
+  * Write a test
+  * Run it ( No implementation code in place, Test needs to fail)
+  * Write implemnetation code to make the test pass
+  * Run it
+  * Refactor the test for better code quality
+  
+This procdure is called **RED-GREEN-REFACTOR**
+
+Final Note: TDD is not only about testing; it is the process of approaching your design and forcing you to think about the desired outcome and API before you code.
```
Comment (mlvandijk) [inline docs/tdd-introduction.md]: remove "only"

File: docs/tdd-introduction.md [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+
+To get better understanding here we go life cycle of TDD
+
+  * Write a test
+  * Run it ( No implementation code in place, Test needs to fail)
+  * Write implemnetation code to make the test pass
+  * Run it
+  * Refactor the test for better code quality
+  
+This procdure is called **RED-GREEN-REFACTOR**
```
Comment (mlvandijk) [inline docs/tdd-introduction.md]: "procdure" should be "procedure"

**Comment by mlvandijk** [comment]
@mgundala If you wouldn't mind, could you (also) PR these changes in https://github.com/cucumber/docs.cucumber.io? They'd be a really nice addition to the docs

**Comment by gmk1217** [comment]
On Wed, Oct 25, 2017, 9:10 PM Marit van Dijk <notifications@github.com>
wrote:

> @mgundala <https://github.com/mgundala> If you wouldn't mind, could you
> (also) PR these changes in https://github.com/cucumber/docs.cucumber.io?
> They'd be a really nice addition to the docs
>
Sure ,will raise One.

**Comment by mlvandijk** [comment]
Closing this PR as these changes have been added to the docs here: https://github.com/cucumber/docs.cucumber.io/pull/53
